src/ike/main.py

```diff
@@ -234,7 +237,52 @@ def dev():
 
 @app.command()
 def deploy():
-    print("I fight for my friends")
+    api_key = keyring.get_password('ike', 'api_key')
+
+    if not api_key:
+        logger.error("API key not found. Please configure with ike config.")
+        typer.Exit(1)
+
+    project_root = os.path.join(os.getcwd(), "docs/")
+    node_root = _get_node_root(project_root)
+    
+    logger.info("Queueing deployment...")
+    _zip_build(node_root)
+    _submit_deployment(api_key)
+
+    
+def _zip_build(node_root: str):
+    with zipfile.ZipFile("build.zip", 'w', zipfile.ZIP_DEFLATED) as zipf:
+        for root, _, files in os.walk(node_root):
+            if "node_modules" in root:
+                continue  # Exclude node_modules from zip
+            
+            for file in files:
+                file_path = os.path.join(root, file)
+                zipf.write(file_path, os.path.relpath(file_path, node_root))
+
+
+def _submit_deployment(api_key: str) -> str:
+    with open("build.zip", "rb") as file:
+        response = requests.post("https://yron03hrwk.execute-api.us-east-1.amazonaws.com/dev/docs/build", 
+            headers={
+                "x-api-key": f"{api_key}",
```

Comment:
I assume `api_key` is a `str`?
```suggestion
                "x-api-key": api_key,
```